Source/Scene/Cesium3DTileset.js

```diff
@@ -1246,6 +1332,101 @@ define([
         return flag;
     }
 
+    var descendantStack = [];
+
+    function markNearestLoadedTilesForSelection(tileset, frameState, selectionState, outOfCore) {
+        var finalQueue = selectionState.finalQueue;
+        var selectionQueue = selectionState.selectionQueue;
+        selectionQueue.length = 0;
+        var frameNumber = frameState.frameNumber;
+
+        var length = finalQueue.length;
+        for (var i = 0; i < length; ++i) {
+            var original = finalQueue.get(i);
+            var tile = original;
+            // traverse up the tree to find a ready ancestor
+            if (tile.hasContent || tile.hasTilesetContent) {  // could be Empty3DTileContent
+                while (defined(tile) && !(tile.hasContent && tile.contentReady)) {
+                    if (!tile.contentReady) {
+                        tileset._hasMixedContent = true;
+                    }
+                    tile = tile.parent;
+                }
+            }
+
+            if (defined(tile)) {
+                if (!tile.selected) {
+                    tile._finalResolution = (tile === original || tile.refine === Cesium3DTileRefine.ADD);
+                    tile.selected = true;
+                    tile._selectedFrame = frameNumber;
+                    selectionQueue.push(tile);
+                }
+            } else {
+                // if no ancestors are ready, traverse down and select ready tiles to minimize empty regions
+                descendantStack.push(original);
```

Comment:
Yea, it's still needed. I just added a condition to limit it to traversing down so deep tilesets don't become a problem